src/kudu/consensus/raft_consensus.cc

```diff
@@ -2389,10 +2390,19 @@ Status RaftConsensus::CheckBulkConfigChangeAndGetNewConfigUnlocked(
             return Status::InvalidArgument("peer must have last_known_addr specified",
                                            SecureShortDebugString(req));
           }
-          if (peer.member_type() != RaftPeerPB::VOTER && peer.has_attrs() &&
-              peer.attrs().has_quorum_id() && peer.attrs().quorum_id() != "") {
-            return Status::InvalidArgument("Non-voter must not have quorum_id",
+          if (peer.has_member_type() && peer.member_type() == RaftPeerPB::VOTER) {
+            if (FLAGS_enable_flexi_raft && IsUseQuorumId(committed_config.commit_rule())) {
+              if (!PeerHasNonEmptyQuorumId(peer)) {
```

Comment:
nit: Combine this if with the one above

Reply:
It is combined with the one above ---- in the else clause.